In [54]:
import pandas as pd

In [56]:
modulo1 = pd.read_csv('Enaho01-2020-100.csv')

In [57]:
modulo341 = pd.read_csv('Sumaria-2020.csv')

2. - ¿Cuáles son las dimensiones de la sumaria? (o el número de filas y columnas).
   - ¿Cuáles son las dimensiones del módulo de vivienda? (o el número de filas y columnas).

In [58]:
modulo1.shape 

(53423, 331)

In [107]:
modulo341.shape

(34490, 182)

In [108]:
modulo341

A�O  MES  CONGLOME  VIVIENDA  HOGAR  UBIGEO  DOMINIO  ESTRATO  \
0      2020    1      5009        12     11   10101        4        4   
1      2020    1      5009        41     11   10101        4        4   
2      2020    1      5009        98     11   10101        4        4   
3      2020    1      5015         7     11   10101        4        4   
4      2020    1      5015        19     11   10101        4        4   
...     ...  ...       ...       ...    ...     ...      ...      ...   
34485  2020   12     20305        31     11  250301        7        4   
34486  2020   12     20305        52     11  250301        7        4   
34487  2020   12     20305        73     11  250301        7        4   
34488  2020   12     20305       116     11  250301        7        4   
34489  2020   12     20305       137     11  250301        7        4   

       MIEPERHO  TOTMIEHO  ...  TIPOENTREVISTA    FACTOR07  NCONGLOME  \
0             2         2  ...               1  149.202744       7076   
1             2         2  ...               1  149.202744       7076   
2             6         6  ...               1  149.202744       7076   
3             4         4  ...               1   70.492592       7089   
4             4         4  ...               1   70.492592       7089   
...         ...       ...  ...             ...         ...        ...   
34485         1         1  ...               1  275.828552      15122   
34486         1         1  ...               1  275.828552      15122   
34487         1         1  ...               1  275.828552      15122   
34488         4         4  ...               1  275.828552      15122   
34489         4         4  ...               1  275.828552      15122   

       SUB_CONGLOME  LINEAV_RPL      LINEAV  POBREZAV  DEPENDIENTES  \
0                 0         657  560.468506         4             0   
1                 0         657  553.731506         4             0   
2                 0         657  547.194031         3             2   
3                 0         657  556.025330         4             1   
4                 0         657  548.526917         4             1   
...             ...         ...         ...       ...           ...   
34485             0         657  641.312134         4             0   
34486             0         657  642.814026         4             0   
34487             0         657  641.782898         4             0   
34488             0         657  643.160034         3             3   
34489             0         657  639.233521         3             2   

       INGRESOxMES  INGRESO-GASTO  
0      1553.101074   12208.628906  
1      1748.648763   13317.550781  
2       881.094482    4156.316406  
3      1051.839193   11054.056641  
4      2546.721137   28370.421875  
...            ...            ...  
34485  1185.662598    1230.478516  
34486   953.500000     751.404297  
34487  1070.583333    1270.196289  
34488  1750.779460   -2769.675781  
34489   825.943766    2949.781250  

[34490 rows x 182 columns]

3. En esta oportunidad no necesitaremos todas las variables de la ENAHO para hacer esta tarea. 
- Quedémonos con las variables ``` CONGLOME VIVIENDA HOGAR MIEPERHO PERCEPHO POBREZA GASHOG2D INGHOG2D ``` de la sumaria. 
- Quedémonos con las variables ``` CONGLOME VIVIENDA HOGAR RESULT NBI1 NBI2 NBI3 NBI4 NBI5 ``` del módulo de vivienda.

 
 Nota: cada una de las variables es:
- MIEPERHO: número de miembros del hogar. 
- PERCEPHO: número de perceptores del hogar (que reciben un ingreso). 
- POBREZA: Pobreza monetaria según la línea de pobreza. 
- GASHOG2D: Gasto bruto anual del hogar. 
- INGHOG2D: Ingreso anual neto total del hogar. 

Necesidades básicas:
- NBI1: vivienda inadecuada.
- NBI2: hacinamiento
- NBI3: vivienda sin servicios higiénicos
- NBI4: hogares con  niños que no asisten a la escuela.
- NBI5: hogares con alta dependencia económica 

CONGLOME, VIVIENDA y HOGAR son los identificadores únicos a nivel de hogar (como un key id).

In [109]:
needed1 = ['CONGLOME', 'VIVIENDA', 'HOGAR', 'RESULT', 'NBI1', 'NBI2', 'NBI4', 'NBI5']

In [110]:
needed341 = ['CONGLOME', 'VIVIENDA', 'HOGAR', 'MIEPERHO', 'PERCEPHO', 'POBREZA', 'GASHOG2D', 'INGHOG2D']

In [111]:
modulo1[needed1]

CONGLOME  VIVIENDA  HOGAR  RESULT NBI1 NBI2 NBI4 NBI5
0          5009        12     11       1    0    0    0    0
1          5009        41     11       1    0    0    0    0
2          5009        56     11       7                    
3          5009        84     11       7                    
4          5009        98     11       1    0    0    0    0
...         ...       ...    ...     ...  ...  ...  ...  ...
53418     20305        52     11       1    0    0    0    0
53419     20305        73     11       1    1    0    0    0
53420     20305        94     11       7                    
53421     20305       116     11       1    0    0    0    0
53422     20305       137     11       1    0    0    0    0

[53423 rows x 8 columns]

In [112]:
modulo341_small = modulo341[needed341]

In [113]:
modulo1_small = modulo1[needed1]

In [114]:
modulo1_small

CONGLOME  VIVIENDA  HOGAR  RESULT NBI1 NBI2 NBI4 NBI5
0          5009        12     11       1    0    0    0    0
1          5009        41     11       1    0    0    0    0
2          5009        56     11       7                    
3          5009        84     11       7                    
4          5009        98     11       1    0    0    0    0
...         ...       ...    ...     ...  ...  ...  ...  ...
53418     20305        52     11       1    0    0    0    0
53419     20305        73     11       1    1    0    0    0
53420     20305        94     11       7                    
53421     20305       116     11       1    0    0    0    0
53422     20305       137     11       1    0    0    0    0

[53423 rows x 8 columns]

In [115]:
modulo341_small

CONGLOME  VIVIENDA  HOGAR  MIEPERHO  PERCEPHO  POBREZA      GASHOG2D  \
0          5009        12     11         2         2        3  25065.796875   
1          5009        41     11         2         2        3  28650.019531   
2          5009        98     11         6         4        3  38136.218750   
3          5015         7     11         4         3        3  26812.154297   
4          5015        19     11         4         3        3  63311.539062   
...         ...       ...    ...       ...       ...      ...           ...   
34485     20305        31     11         1         1        3  12997.472656   
34486     20305        52     11         1         1        3  10690.595703   
34487     20305        73     11         1         1        3  11576.803711   
34488     20305       116     11         4         1        3  23779.029297   
34489     20305       137     11         4         2        3  16872.869141   

           INGHOG2D  
0      37274.425781  
1      41967.570312  
2      42292.535156  
3      37866.210938  
4      91681.960938  
...             ...  
34485  14227.951172  
34486  11442.000000  
34487  12847.000000  
34488  21009.353516  
34489  19822.650391  

[34490 rows x 8 columns]

4. El módulo vivienda incluye todos los hogares que entraron en el marco muestral, incluso aquellos que no participaron en la ENAHO por diversos motivos. Para eliminar estas viviendas, quedémonos con aquellas cuyo ``` RESULT``` sea igual a 1 ó 2. Almacena este resultado para después.

Type Markdown and LaTeX:  𝛼2

In [116]:
cond1 = modulo1_small['RESULT'] >= 1

cond2 = modulo1_small['RESULT'] <= 2


modulo1_smallr = modulo1_small[cond1 & cond2]

modulo1_smallr

CONGLOME  VIVIENDA  HOGAR  RESULT NBI1 NBI2 NBI4 NBI5
0          5009        12     11       1    0    0    0    0
1          5009        41     11       1    0    0    0    0
4          5009        98     11       1    0    0    0    0
6          5015         7     11       1    0    0    0    0
7          5015        19     11       1    0    0    0    0
...         ...       ...    ...     ...  ...  ...  ...  ...
53417     20305        31     11       1    0    0    0    0
53418     20305        52     11       1    0    0    0    0
53419     20305        73     11       1    1    0    0    0
53421     20305       116     11       1    0    0    0    0
53422     20305       137     11       1    0    0    0    0

[34490 rows x 8 columns]

5. Queremos calcular una serie de variables (ó columnas nuevas):  
En sumaria: 
- Queremos el número de miembros dependientes del hogar . Estos son el número de personas que no perciben un ingreso ó la diferencia entre el total de miembros y el total de perceptores. 
- Queremos el ingreso mensual promedio por perceptor del hogar.
- Queremos la diferencia  entre ingreso y gasto del hogar


MIEPERHO: número de miembros del hogar.
PERCEPHO: número de perceptores del hogar (que reciben un ingreso).
POBREZA: Pobreza monetaria según la línea de pobreza.
GASHOG2D: Gasto bruto anual del hogar.
INGHOG2D: Ingreso anual neto total del hogar.

In [117]:
personas_dep = modulo341['MIEPERHO'] - modulo341['PERCEPHO']

modulo341['DEPENDIENTES'] = personas_dep

modulo341

A�O  MES  CONGLOME  VIVIENDA  HOGAR  UBIGEO  DOMINIO  ESTRATO  \
0      2020    1      5009        12     11   10101        4        4   
1      2020    1      5009        41     11   10101        4        4   
2      2020    1      5009        98     11   10101        4        4   
3      2020    1      5015         7     11   10101        4        4   
4      2020    1      5015        19     11   10101        4        4   
...     ...  ...       ...       ...    ...     ...      ...      ...   
34485  2020   12     20305        31     11  250301        7        4   
34486  2020   12     20305        52     11  250301        7        4   
34487  2020   12     20305        73     11  250301        7        4   
34488  2020   12     20305       116     11  250301        7        4   
34489  2020   12     20305       137     11  250301        7        4   

       MIEPERHO  TOTMIEHO  ...  TIPOENTREVISTA    FACTOR07  NCONGLOME  \
0             2         2  ...               1  149.202744       7076   
1             2         2  ...               1  149.202744       7076   
2             6         6  ...               1  149.202744       7076   
3             4         4  ...               1   70.492592       7089   
4             4         4  ...               1   70.492592       7089   
...         ...       ...  ...             ...         ...        ...   
34485         1         1  ...               1  275.828552      15122   
34486         1         1  ...               1  275.828552      15122   
34487         1         1  ...               1  275.828552      15122   
34488         4         4  ...               1  275.828552      15122   
34489         4         4  ...               1  275.828552      15122   

       SUB_CONGLOME  LINEAV_RPL      LINEAV  POBREZAV  DEPENDIENTES  \
0                 0         657  560.468506         4             0   
1                 0         657  553.731506         4             0   
2                 0         657  547.194031         3             2   
3                 0         657  556.025330         4             1   
4                 0         657  548.526917         4             1   
...             ...         ...         ...       ...           ...   
34485             0         657  641.312134         4             0   
34486             0         657  642.814026         4             0   
34487             0         657  641.782898         4             0   
34488             0         657  643.160034         3             3   
34489             0         657  639.233521         3             2   

       INGRESOxMES  INGRESO-GASTO  
0      1553.101074   12208.628906  
1      1748.648763   13317.550781  
2       881.094482    4156.316406  
3      1051.839193   11054.056641  
4      2546.721137   28370.421875  
...            ...            ...  
34485  1185.662598    1230.478516  
34486   953.500000     751.404297  
34487  1070.583333    1270.196289  
34488  1750.779460   -2769.675781  
34489   825.943766    2949.781250  

[34490 rows x 182 columns]

In [118]:
ingreso_mensual = modulo341_small['INGHOG2D'] / modulo341_small['PERCEPHO'] / 12 

modulo341['INGRESOxMES'] = ingreso_mensual

modulo341

A�O  MES  CONGLOME  VIVIENDA  HOGAR  UBIGEO  DOMINIO  ESTRATO  \
0      2020    1      5009        12     11   10101        4        4   
1      2020    1      5009        41     11   10101        4        4   
2      2020    1      5009        98     11   10101        4        4   
3      2020    1      5015         7     11   10101        4        4   
4      2020    1      5015        19     11   10101        4        4   
...     ...  ...       ...       ...    ...     ...      ...      ...   
34485  2020   12     20305        31     11  250301        7        4   
34486  2020   12     20305        52     11  250301        7        4   
34487  2020   12     20305        73     11  250301        7        4   
34488  2020   12     20305       116     11  250301        7        4   
34489  2020   12     20305       137     11  250301        7        4   

       MIEPERHO  TOTMIEHO  ...  TIPOENTREVISTA    FACTOR07  NCONGLOME  \
0             2         2  ...               1  149.202744       7076   
1             2         2  ...               1  149.202744       7076   
2             6         6  ...               1  149.202744       7076   
3             4         4  ...               1   70.492592       7089   
4             4         4  ...               1   70.492592       7089   
...         ...       ...  ...             ...         ...        ...   
34485         1         1  ...               1  275.828552      15122   
34486         1         1  ...               1  275.828552      15122   
34487         1         1  ...               1  275.828552      15122   
34488         4         4  ...               1  275.828552      15122   
34489         4         4  ...               1  275.828552      15122   

       SUB_CONGLOME  LINEAV_RPL      LINEAV  POBREZAV  DEPENDIENTES  \
0                 0         657  560.468506         4             0   
1                 0         657  553.731506         4             0   
2                 0         657  547.194031         3             2   
3                 0         657  556.025330         4             1   
4                 0         657  548.526917         4             1   
...             ...         ...         ...       ...           ...   
34485             0         657  641.312134         4             0   
34486             0         657  642.814026         4             0   
34487             0         657  641.782898         4             0   
34488             0         657  643.160034         3             3   
34489             0         657  639.233521         3             2   

       INGRESOxMES  INGRESO-GASTO  
0      1553.101074   12208.628906  
1      1748.648763   13317.550781  
2       881.094482    4156.316406  
3      1051.839193   11054.056641  
4      2546.721137   28370.421875  
...            ...            ...  
34485  1185.662598    1230.478516  
34486   953.500000     751.404297  
34487  1070.583333    1270.196289  
34488  1750.779460   -2769.675781  
34489   825.943766    2949.781250  

[34490 rows x 182 columns]

In [119]:

dif_ingreso_gas = (modulo341_small['INGHOG2D'] - modulo341_small['GASHOG2D'])

modulo341['INGRESO-GASTO'] = dif_ingreso_gas

modulo341

A�O  MES  CONGLOME  VIVIENDA  HOGAR  UBIGEO  DOMINIO  ESTRATO  \
0      2020    1      5009        12     11   10101        4        4   
1      2020    1      5009        41     11   10101        4        4   
2      2020    1      5009        98     11   10101        4        4   
3      2020    1      5015         7     11   10101        4        4   
4      2020    1      5015        19     11   10101        4        4   
...     ...  ...       ...       ...    ...     ...      ...      ...   
34485  2020   12     20305        31     11  250301        7        4   
34486  2020   12     20305        52     11  250301        7        4   
34487  2020   12     20305        73     11  250301        7        4   
34488  2020   12     20305       116     11  250301        7        4   
34489  2020   12     20305       137     11  250301        7        4   

       MIEPERHO  TOTMIEHO  ...  TIPOENTREVISTA    FACTOR07  NCONGLOME  \
0             2         2  ...               1  149.202744       7076   
1             2         2  ...               1  149.202744       7076   
2             6         6  ...               1  149.202744       7076   
3             4         4  ...               1   70.492592       7089   
4             4         4  ...               1   70.492592       7089   
...         ...       ...  ...             ...         ...        ...   
34485         1         1  ...               1  275.828552      15122   
34486         1         1  ...               1  275.828552      15122   
34487         1         1  ...               1  275.828552      15122   
34488         4         4  ...               1  275.828552      15122   
34489         4         4  ...               1  275.828552      15122   

       SUB_CONGLOME  LINEAV_RPL      LINEAV  POBREZAV  DEPENDIENTES  \
0                 0         657  560.468506         4             0   
1                 0         657  553.731506         4             0   
2                 0         657  547.194031         3             2   
3                 0         657  556.025330         4             1   
4                 0         657  548.526917         4             1   
...             ...         ...         ...       ...           ...   
34485             0         657  641.312134         4             0   
34486             0         657  642.814026         4             0   
34487             0         657  641.782898         4             0   
34488             0         657  643.160034         3             3   
34489             0         657  639.233521         3             2   

       INGRESOxMES  INGRESO-GASTO  
0      1553.101074   12208.628906  
1      1748.648763   13317.550781  
2       881.094482    4156.316406  
3      1051.839193   11054.056641  
4      2546.721137   28370.421875  
...            ...            ...  
34485  1185.662598    1230.478516  
34486   953.500000     751.404297  
34487  1070.583333    1270.196289  
34488  1750.779460   -2769.675781  
34489   825.943766    2949.781250  

[34490 rows x 182 columns]

In [120]:
modulo342 = modulo341.copy(deep = True)
modulo342

A�O  MES  CONGLOME  VIVIENDA  HOGAR  UBIGEO  DOMINIO  ESTRATO  \
0      2020    1      5009        12     11   10101        4        4   
1      2020    1      5009        41     11   10101        4        4   
2      2020    1      5009        98     11   10101        4        4   
3      2020    1      5015         7     11   10101        4        4   
4      2020    1      5015        19     11   10101        4        4   
...     ...  ...       ...       ...    ...     ...      ...      ...   
34485  2020   12     20305        31     11  250301        7        4   
34486  2020   12     20305        52     11  250301        7        4   
34487  2020   12     20305        73     11  250301        7        4   
34488  2020   12     20305       116     11  250301        7        4   
34489  2020   12     20305       137     11  250301        7        4   

       MIEPERHO  TOTMIEHO  ...  TIPOENTREVISTA    FACTOR07  NCONGLOME  \
0             2         2  ...               1  149.202744       7076   
1             2         2  ...               1  149.202744       7076   
2             6         6  ...               1  149.202744       7076   
3             4         4  ...               1   70.492592       7089   
4             4         4  ...               1   70.492592       7089   
...         ...       ...  ...             ...         ...        ...   
34485         1         1  ...               1  275.828552      15122   
34486         1         1  ...               1  275.828552      15122   
34487         1         1  ...               1  275.828552      15122   
34488         4         4  ...               1  275.828552      15122   
34489         4         4  ...               1  275.828552      15122   

       SUB_CONGLOME  LINEAV_RPL      LINEAV  POBREZAV  DEPENDIENTES  \
0                 0         657  560.468506         4             0   
1                 0         657  553.731506         4             0   
2                 0         657  547.194031         3             2   
3                 0         657  556.025330         4             1   
4                 0         657  548.526917         4             1   
...             ...         ...         ...       ...           ...   
34485             0         657  641.312134         4             0   
34486             0         657  642.814026         4             0   
34487             0         657  641.782898         4             0   
34488             0         657  643.160034         3             3   
34489             0         657  639.233521         3             2   

       INGRESOxMES  INGRESO-GASTO  
0      1553.101074   12208.628906  
1      1748.648763   13317.550781  
2       881.094482    4156.316406  
3      1051.839193   11054.056641  
4      2546.721137   28370.421875  
...            ...            ...  
34485  1185.662598    1230.478516  
34486   953.500000     751.404297  
34487  1070.583333    1270.196289  
34488  1750.779460   -2769.675781  
34489   825.943766    2949.781250  

[34490 rows x 182 columns]

6. La variable de `pobreza`, en la sumaria, está codificada como integers 1, 2 y 3. Esto corresponde a:  
`1`: hogar pobre extremo  
`2`: hogar pobre no extremo  
`3`: hogar no  pobre  

En base a esta variable, crea dos variables más:
1. Una variable que tenga `'pobre extremo'`, `'pobre no extremo'` y `'no pobre'` en vez  de 1, 2, 3.
2. Una variable que tenga `'pobre'` y `'no pobre'` (ó que englobe a los hogares pobres y pobres no extremos).


In [121]:
modulo342['tipo_pobreza'] = modulo342['POBREZA']

diccionario_cambios = {'1': 'hogar pobre extremo',
                       '2': 'hogar pobre no extremo','3': 'hogar no pobre'}

modulo342.replace({'POBREZA' : diccionario_cambios}, inplace = True)

In [122]:
modulo342['tipo_pobreza'].value_counts()


3    26766
2     6272
1     1452
Name: tipo_pobreza, dtype: int64

In [123]:
modulo342.loc[modulo342['POBREZA'] == '1', 
              'POBREZA'] = 'hogar pobre extremo'

modulo342

A�O  MES  CONGLOME  VIVIENDA  HOGAR  UBIGEO  DOMINIO  ESTRATO  \
0      2020    1      5009        12     11   10101        4        4   
1      2020    1      5009        41     11   10101        4        4   
2      2020    1      5009        98     11   10101        4        4   
3      2020    1      5015         7     11   10101        4        4   
4      2020    1      5015        19     11   10101        4        4   
...     ...  ...       ...       ...    ...     ...      ...      ...   
34485  2020   12     20305        31     11  250301        7        4   
34486  2020   12     20305        52     11  250301        7        4   
34487  2020   12     20305        73     11  250301        7        4   
34488  2020   12     20305       116     11  250301        7        4   
34489  2020   12     20305       137     11  250301        7        4   

       MIEPERHO  TOTMIEHO  ...    FACTOR07  NCONGLOME  SUB_CONGLOME  \
0             2         2  ...  149.202744       7076             0   
1             2         2  ...  149.202744       7076             0   
2             6         6  ...  149.202744       7076             0   
3             4         4  ...   70.492592       7089             0   
4             4         4  ...   70.492592       7089             0   
...         ...       ...  ...         ...        ...           ...   
34485         1         1  ...  275.828552      15122             0   
34486         1         1  ...  275.828552      15122             0   
34487         1         1  ...  275.828552      15122             0   
34488         4         4  ...  275.828552      15122             0   
34489         4         4  ...  275.828552      15122             0   

       LINEAV_RPL      LINEAV  POBREZAV  DEPENDIENTES  INGRESOxMES  \
0             657  560.468506         4             0  1553.101074   
1             657  553.731506         4             0  1748.648763   
2             657  547.194031         3             2   881.094482   
3             657  556.025330         4             1  1051.839193   
4             657  548.526917         4             1  2546.721137   
...           ...         ...       ...           ...          ...   
34485         657  641.312134         4             0  1185.662598   
34486         657  642.814026         4             0   953.500000   
34487         657  641.782898         4             0  1070.583333   
34488         657  643.160034         3             3  1750.779460   
34489         657  639.233521         3             2   825.943766   

       INGRESO-GASTO  tipo_pobreza  
0       12208.628906             3  
1       13317.550781             3  
2        4156.316406             3  
3       11054.056641             3  
4       28370.421875             3  
...              ...           ...  
34485    1230.478516             3  
34486     751.404297             3  
34487    1270.196289             3  
34488   -2769.675781             3  
34489    2949.781250             3  

[34490 rows x 183 columns]

In [124]:
modulo342['POBREZA'].value_counts()

3    26766
2     6272
1     1452
Name: POBREZA, dtype: int64

7. Queremos saber el promedio de ingresos de los hogares (inghog2d) por nivel de pobreza. Haz un `groupby` para averiguar esto. 


(Por ejemplo, todos los hogares pobres extremos tendrán un único promedio, y así con las 2 categorías de pobreza adicionales). 

In [125]:
modulo341[['INGHOG2D', 'POBREZA']].groupby('POBREZA').mean()

INGHOG2D
POBREZA              
1         9980.110857
2        15565.160905
3        32014.094308

8. ¡Ahora uniremos los datos!  
Pero antes un check de sanidad: Verifica que el número  de filas de ambas bases de datos son las mismas (acuérdate que filtramos las observaciones por `result`).  La ENAHO recolecta información de estos 34,490 hogares a través de todos sus módulos. Para utilizar estos módulos a la vez, realizaremos una "unión" (merge).   

Une ambas bases con un _inner merge_. Recuerda que el id que identifica a cada hogar se compone de ` conglome vivienda hogar`.  

In [126]:
modulo1_smallr.shape

(34490, 8)

In [127]:
modulo341_small.shape

(34490, 8)

In [132]:
modulo1_smallr_1 = modulo1_smallr[['CONGLOME', 'VIVIENDA', 'HOGAR']]

modulo341_small_2 = modulo341_small[['CONGLOME', 'VIVIENDA', 'HOGAR']]

In [133]:

resultado_merge = pd.merge(modulo1_smallr_1,
                     modulo341_small_2,
                    left_on = ['CONGLOME', 'VIVIENDA', 'HOGAR'],
                    right_on = ['CONGLOME', 'VIVIENDA', 'HOGAR'],
                    how = 'inner', indicator = True)
resultado_merge

CONGLOME  VIVIENDA  HOGAR _merge
0          5009        12     11   both
1          5009        41     11   both
2          5009        98     11   both
3          5015         7     11   both
4          5015        19     11   both
...         ...       ...    ...    ...
34485     20305        31     11   both
34486     20305        52     11   both
34487     20305        73     11   both
34488     20305       116     11   both
34489     20305       137     11   both

[34490 rows x 4 columns]

9. Haz un ```groupby``` que agrupe (`agg`) los datos por nivel de pobreza  y la nbi1. Queremos hallar el promedio y la mediana de los ingresos (inghog2d) y los gastos (gashog2d) por el producto cartesiano de las categorías de estas 2 variables - pobreza y nbi 1 -. 

El producto cartesiano es tan sólo todas las posibles combinaciones entre categorías. 
En este caso, tendremos:
- pobre extremo con nbi1
- pobre extremo sin nbi1
- pobre no extremo con nbi1
- pobre no extremo sin nbi1
- no pobre con nbi1
- no pobre sin nbi1

(Este es un  ejemplo de cómo, al cruzar un indicador de pobreza monetaria -pobreza- y pobreza multidimensional-nbis-, podemos seguir observando brechas).